extensions/SwiftCloudFiles/php-cloudfiles-1.7.10/cloudfiles_http.php

```diff
@@ -1483,7 +1483,8 @@ private function _send_request($conn_type, $url_path, $hdrs=NULL, $method="GET",
 		while( $retriesLeft >= 0 ) {
 			$res = $this->_do_send_request( $conn_type, $url_path, $hdrs, $method, $force_new );
 
-			if ( !in_array( $res, [ 500, 503, false ] ) ) {
+			# PLATFORM-1059 - retry all HTTP 50x responses
+			if ( $res !== false && $res < 500 ) {
 				// request was successful, return
```

Comment:
The comment here is slighly misleading. Note, that 4xx indicate client errors. We can't say the request successful on client error. :-)